Index: b/arch/arm/boot/dts/suniv-f1c200s.dtsi
===================================================================
--- a/arch/arm/boot/dts/suniv-f1c200s.dtsi
+++ b/arch/arm/boot/dts/suniv-f1c200s.dtsi
@@ -370,6 +370,12 @@
 				bias-pull-up;
 			};
 
+			uart2_pe_pins: uart2-pe-pins {
+				pins = "PE7", "PE8";
+				function = "uart2";
+				bias-pull-up;
+			};
+
 			csi0_pins: csi0-pins {
 				pins = "PE0", "PE1", "PE2", "PE3", "PE4", "PE5",
 					"PE6","PE7", "PE8", "PE9", "PE10";
