Index: b/scripts/Makefile.build
===================================================================
--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -413,9 +413,9 @@ ifneq ($(userprogs),)
 include $(srctree)/scripts/Makefile.userprogs
 endif
 
-ifneq ($(need-dtbslist)$(dtb-y)$(dtb-)$(filter %.dtb %.dtb.o %.dtbo.o,$(targets)),)
+# ifneq ($(need-dtbslist)$(dtb-y)$(dtb-)$(filter %.dtb %.dtb.o %.dtbo.o,$(targets)),)
 include $(srctree)/scripts/Makefile.dtbs
-endif
+# endif
 
 # Build
 # ---------------------------------------------------------------------------
