Index: b/drivers/net/wireless/Kconfig
===================================================================
--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -36,6 +36,8 @@ source "drivers/net/wireless/st/Kconfig"
 source "drivers/net/wireless/ti/Kconfig"
 source "drivers/net/wireless/zydas/Kconfig"
 source "drivers/net/wireless/quantenna/Kconfig"
+source "drivers/net/wireless/xr819s/Kconfig"
+source "drivers/net/wireless/xr829/Kconfig"
 
 source "drivers/net/wireless/virtual/Kconfig"
 
Index: b/drivers/net/wireless/Makefile
===================================================================
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -21,5 +21,7 @@ obj-$(CONFIG_WLAN_VENDOR_SILABS) += sila
 obj-$(CONFIG_WLAN_VENDOR_ST) += st/
 obj-$(CONFIG_WLAN_VENDOR_TI) += ti/
 obj-$(CONFIG_WLAN_VENDOR_ZYDAS) += zydas/
+obj-$(CONFIG_XR829_WLAN)	+= xr829/
+obj-$(CONFIG_XR819S_WLAN)	+= xr819s/
 
 obj-$(CONFIG_WLAN) += virtual/
