Index: b/drivers/net/ethernet/stmicro/stmmac/dwmac-sun8i.c
===================================================================
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-sun8i.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-sun8i.c
@@ -154,6 +154,20 @@ static const struct emac_variant emac_va
 	.tx_delay_max = 7,
 };
 
+static const struct emac_variant emac_variant_a100 = {
+	.default_syscon_value = 0x51fe6,
+	.syscon_field = &sun8i_syscon_reg_field,
+	/* The "Internal PHY" of H6 is not on the die. It's on the
+	 * co-packaged AC200 chip instead.
+	 */
+	.soc_has_internal_phy = false,
+	.support_mii = false,
+	.support_rmii = true,
+	.support_rgmii = true,
+	.rx_delay_max = 31,
+	.tx_delay_max = 7,
+};
+
 #define EMAC_BASIC_CTL0 0x00
 #define EMAC_BASIC_CTL1 0x04
 #define EMAC_INT_STA    0x08
@@ -1337,6 +1351,8 @@ static const struct of_device_id sun8i_d
 		.data = &emac_variant_a64 },
 	{ .compatible = "allwinner,sun50i-h6-emac",
 		.data = &emac_variant_h6 },
+	{ .compatible = "allwinner,sun50i-a100-emac",
+		.data = &emac_variant_a100 },
 	{ }
 };
 MODULE_DEVICE_TABLE(of, sun8i_dwmac_match);
